### ISSUE TITLE
Microsoft.UI.Xaml.Controls.TextBox.CharacterCasing is not implemented in Uno

### ISSUE THREAD

**Issue by cesarchefinho** [unoplatform/uno#8634]
### Current behavior

Not working on webassembly, works on all other platforms

### Expected behavior

control casing of texbox

### Works on UWP/WinUI

Yes

### Environment

Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia

### NuGet package version(s)

4.2.6

### Affected platforms

WebAssembly

### IDE

Visual Studio 2022

**Comment by iury-kc** [comment]
Android already had a method called UpdateCasing that changes the data on the TextBoxView. But it just works applying filters for user input data. 

I implemented the same way in all platforms, now correcting the casing as it need to do.
However, i could saw by debugging it that the class TextBox (on almost all platforms)  changes the data on the visible part so many times. I'm not sure it it's necessary.
Maybe the oposite way would help: instead TextBox class changing the value of the visual part, this last class (it is not the same class throught the platforms) could be ask for the correct casing version of the TextBox.Text. 
Maybe a internal property or even a method returning the correct string. But it susppect it will affect hardly the component. 

I'm posting the PR as a draft for your consideration.

**Comment by iury-kc** [comment]
BTW, besides having the following on [Microsoft Docs](https://docs.microsoft.com/en-us/windows/winui/api/microsoft.ui.xaml.controls.textbox.charactercasing?view=winui-3.0), on UWP this property also modifies characters already existing on the textbox.

"A value of the enumeration that indicates how the control modifies the case of characters as they are typed. The default is Normal, which indicates that the charcters are not changed."

**Comment by cesarchefinho** [comment]
does #12109 resolve this PR issues ?

**Comment by jeromelaban** [comment]
@cesarchefinho it's not supposed to. Are you talking about https://github.com/unoplatform/uno/pull/8658 ?

**Comment by cesarchefinho** [comment]
Yes. The PR dont wotk at test. maybe the problem is the test and not the PR.

**Comment by jeromelaban** [comment]
It's likely the test, though the PR is quite old and needs to be rebased.

**Comment by MartinZikmund** [comment]
Still relevant for Skia targets and should be relatively easy to add now

**Comment by MartinZikmund** [comment]
# Repro generator

This issue was classified as not reproducible with a code sample or test.

**Reason:** WebAssembly-specific bug — TextBox.CharacterCasing reportedly works on all other platforms but not WASM.
